core/src/main/java/org/gbif/pipelines/core/functions/interpretation/DayInterpreter.java

```diff
@@ -0,0 +1,58 @@
+package org.gbif.pipelines.core.functions.interpretation;
+
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueType;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.core.functions.interpretation.error.LineageType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * interprets day and adds issues and lineages with it
+ */
+public class DayInterpreter implements Interpretable<String, Integer> {
+
+  private static final int min_day = 1;
+  private static final int max_day = 31;
+
+  @Override
+  public Integer interpret(String input) throws InterpretationException {
+    Integer x;
+    List<Issue> issues = new ArrayList<>();
+    List<Lineage> lineages = new ArrayList<>();
+
+    try {
+      x = Integer.parseInt(input);
+    } catch (NumberFormatException ex) {
+      //if parse failed
+      issues.add(Issue.newBuilder()
+                   .setRemark("Day cannot be parsed because of " + ex.getMessage())
+                   .setIssueType(IssueType.OTHERS)
+                   .build());
+      lineages.add(Lineage.newBuilder()
+                     .setRemark("Since day cannot be parsed setting it to null")
+                     .setLineageType(LineageType.SET_TO_NULL)
+                     .build());
+      InterpretationException iex = new InterpretationException(issues, lineages);
```

Comment:
the next three lines have been repeated several time in other places, maybe we should move that pattern to the InterpretationException class?